wrlinux/wrlinux.go

```diff
@@ -0,0 +1,273 @@
+	"github.com/araddon/dateparse"
+	"golang.org/x/xerrors"
+	"github.com/aquasecurity/vuln-list-update/utils"
+)
+
+const (
+	cveTrackerDir = "windriver-cve-tracker"
+	windriverDir  = "wrlinux"
+)
+
+var (
+	repoURLs = []string{
+		"https://distro.windriver.com/git/windriver-cve-tracker.git",
+	}
+	targets = []string{
+		"active",
+	}
+	statuses = []string{
+		"released",
+		"pending",
+		"not-affected",
+		"ignored",
+	}
+)
+
+type Vulnerability struct {
+	Candidate         string
+	PublicDate        time.Time
+	Description       string
+	References        []string
+	Notes             []string
+	Priority          string
+	Bugs              []string
+	Patches           map[Package]Statuses
+}
+
+type Package string
+
+type Release string
+
+type Statuses map[Release]Status
+
+type Status struct {
+	Status string
+	Note   string
+}
+
+func Update() error {
+	var err error
+	gc := git.Config{}
+	dir := filepath.Join(utils.CacheDir(), cveTrackerDir)
+	for _, url := range repoURLs {
+		_, err = gc.CloneOrPull(url, dir, "master", false)
+		if err == nil {
+			break
+		}
+		log.Printf("failed to clone or pull: %s: %v", url, err)
+		log.Printf("removing %s directory", cveTrackerDir)
+		if err := os.RemoveAll(dir); err != nil {
+			return xerrors.Errorf("failed to remove %s directory: %w", cveTrackerDir, err)
+		}
+	}
+	if err != nil {
+		return xerrors.Errorf("failed to clone or pull: %w", err)
+	}
+
+	dst := filepath.Join(utils.VulnListDir(), windriverDir)
+	log.Printf("removing windriver directory %s", dst)
+	if err := os.RemoveAll(dst); err != nil {
+		return xerrors.Errorf("failed to remove windriver directory: %w", err)
+	}
+
+	log.Println("walking windriver-cve-tracker ...")
+	for _, target := range targets {
+		if err := walkDir(filepath.Join(dir, target)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func walkDir(root string) error {
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return xerrors.Errorf("file walk error: %w", err)
+		}
+		if info.IsDir() {
+			return nil
+		}
+
+		base := filepath.Base(path)
+		if !strings.HasPrefix(base, "CVE-") {
+			return nil
+		}
+
+		f, err := os.Open(path)
+		if err != nil {
+			return xerrors.Errorf("error in file open: %w", err)
+		}
+
+		vuln, err := parse(f)
+		if err != nil {
+			return xerrors.Errorf("error in parse: %w", err)
+		}
+
+		if err = utils.SaveCVEPerYear(filepath.Join(utils.VulnListDir(), windriverDir), vuln.Candidate, vuln); err != nil {
+			return xerrors.Errorf("error in save: %w", err)
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return xerrors.Errorf("error in walk: %w", err)
+	}
+	return nil
+}
+
+func parse(r io.Reader) (vuln *Vulnerability, err error) {
+	vuln = &Vulnerability{}
+	vuln.Patches = map[Package]Statuses{}
+
+	all, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+	lines := strings.Split(string(all), "\n")
+
+	for i := 0; i < len(lines); i++ {
```

Comment:
We try to use same logic, if it is possible.
Can you use `bufio` here:
https://github.com/aquasecurity/vuln-list-update/blob/728ebaea9cd3a5d3d5770d31364fc198cafc077b/debian/tracker/debian.go#L177-L179

Reply:
I can use bufio to read the file, but I still have to read in the whole file and iterate over it the same way since I have inner loops which modify iterator "i".